### PR TITLE
LF-4480: Add new animal movement task route

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

- Create GET `/animal_movement_purposes` API
- Add `animal_movement_task` to the middleware
- Add routes for POST and PATCH
- Extract some reusable functions from `task.test.js`
- In `testEnvironment`, move `DELETE FROM "location";` after deleting animal data now that animals and batches have `location_id`
- Add tests (`npm run test animalTask.test`)
- Fix typo Joyce found in the previous PR (I thought I fixed in the previous PR...)

Also, I added a middleware to `/patch_due_date/:task_id` not to allow a date earlier than the animals' birth or brought-in dates.

Jira link: https://lite-farm.atlassian.net/browse/LF-4480

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [x] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [x] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Review by antsgar — DISMISSED** [review]
Looks great, nice job with the tests!! 🥳 

Would you mind merging the integration branch into this one? I'm branching off of this for my Task Details PR but my branch includes changes in integration, so if we keep this one up to date with integration I can set this one as base for my PR.

File: packages/api/tests/animalTask.test.js [hunk 1]
```diff
@@ -0,0 +1,996 @@
...
+
+    const checkValidPostRequest = async (type, postData) => {
+      const res = await postTaskRequest({ user_id, farm_id }, type, postData);
+      expect(res.status).toBe(201);
+      const { task_id } = res.body;
+      const createdTask = await knex('task').where({ task_id }).first();
+      expect(createdTask).toBeDefined();
+      const location = await knex('location_tasks').where({ task_id }).first();
+      expect(location.location_id).toBe(postData.locations[0].location_id);
+      expect(location.task_id).toBe(task_id);
```
Comment (Duncan-Brain) [inline packages/api/tests/animalTask.test.js:290]: Is this necessary considering the get query 2 lines above?

File: packages/api/tests/animalTask.test.js [hunk 2]
```diff
@@ -0,0 +1,996 @@
...
+      const specificTask = await knex(type).where({ task_id });
+      expect(specificTask.length).toBe(1);
+      expect(specificTask[0].task_id).toBe(task_id);
```
Comment (Duncan-Brain) [inline packages/api/tests/animalTask.test.js:292]: Ditto.

**Review by Duncan-Brain — DISMISSED** [review]
Looks great ! So many tests ! And I like how you are grouping common validation tests.

Left 1 or 2 inline comments.

File: packages/api/tests/animalTask.test.js (hunk 2 above)
Comment (SayakaOno) [reply packages/api/tests/animalTask.test.js]: Thank you Duncan! 🙏

**Review by Duncan-Brain — APPROVED** [review]

**Review by antsgar — APPROVED** [review]

**Pull request merged**